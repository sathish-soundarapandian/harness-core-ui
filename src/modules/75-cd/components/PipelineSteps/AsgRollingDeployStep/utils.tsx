/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import * as Yup from 'yup'
import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
import type { UseStringsReturn } from 'framework/strings'

export const minimumHealthyPercentageSchema = (
  getString: UseStringsReturn['getString']
): Yup.Schema<number | undefined> => {
  return Yup.lazy((value: number): any => {
    /* istanbul ignore else */
    if (getMultiTypeFromValue(value as any) === MultiTypeInputType.FIXED) {
      return Yup.mixed().test({
        test(): boolean | Yup.ValidationError {
          if (!value) {
            return true
          }
          if (value > 100) {
            return this.createError({ message: getString('cd.minimumHealthyPercentageMaxLimit') })
          }
          if (value < 0) {
            return this.createError({ message: getString('cd.minimumHealthyPercentageMinLimit') })
          }
          return true
        }
      })
    }
    return Yup.number()
  })
}

export const instanceWarmupSchema = (getString: UseStringsReturn['getString']): Yup.Schema<number | undefined> => {
  return Yup.lazy((value: number): any => {
    /* istanbul ignore else */
    if (getMultiTypeFromValue(value as any) === MultiTypeInputType.FIXED) {
      return Yup.mixed().test({
        test(): boolean | Yup.ValidationError {
          if (!value) {
            return true
          }
          if (value < 0) {
            return this.createError({ message: getString('cd.instanceWarmupError') })
          }
          return true
        }
      })
    }
    return Yup.number()
  })
}
