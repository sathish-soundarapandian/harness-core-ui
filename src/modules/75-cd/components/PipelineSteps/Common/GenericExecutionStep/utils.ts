/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { FormikErrors, yupToFormErrors } from 'formik'
import * as Yup from 'yup'
import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'

import type { StepElementConfig } from 'services/cd-ng'
import { getDurationValidationSchema } from '@common/components/MultiTypeDuration/MultiTypeDuration'
import { ValidateInputSetProps, StepViewType } from '@pipeline/components/AbstractSteps/Step'

export const validateGenericFields = ({
  data,
  template,
  getString,
  viewType
}: ValidateInputSetProps<StepElementConfig>): FormikErrors<StepElementConfig> => {
  const isRequired = viewType === StepViewType.DeploymentForm || viewType === StepViewType.TriggerForm
  const errors: FormikErrors<StepElementConfig> = {}

  if (getMultiTypeFromValue(template?.timeout) === MultiTypeInputType.RUNTIME) {
    let timeoutSchema = getDurationValidationSchema({ minimum: '10s' })
    if (isRequired) {
      timeoutSchema = timeoutSchema.required(getString?.('validation.timeout10SecMinimum'))
    }
    const timeout = Yup.object().shape({
      timeout: timeoutSchema
    })
    try {
      timeout.validateSync(data)
    } catch (e) {
      if (e instanceof Yup.ValidationError) {
        const err = yupToFormErrors(e)
        Object.assign(errors, err)
      }
    }
  }
  return errors
}
