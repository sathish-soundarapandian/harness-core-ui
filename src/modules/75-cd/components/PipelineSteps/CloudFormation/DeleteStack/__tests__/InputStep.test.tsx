/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, queryByAttribute, act, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Formik, FormikForm, MultiTypeInputType, RUNTIME_INPUT_VALUE } from '@harness/uicore'
import { TestWrapper } from '@common/utils/testUtils'
import { StepViewType } from '@pipeline/components/AbstractSteps/Step'
import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
import * as cdServices from 'services/cd-ng'
import * as Portal from 'services/portal'
import { DeleteStackTypes, StoreTypes } from '../../CloudFormationInterfaces.types'

import DeleteStackInputStepRef from '../DeleteStackInputSteps'

const initialValues = {
  type: StepType.CloudFormationDeleteStack,
  name: 'test name',
  identifier: 'test_identifier',
  timeout: '10m',
  spec: {
    configuration: {
      type: DeleteStackTypes.Inline,
      spec: {
        provisionerIdentifier: RUNTIME_INPUT_VALUE,
        connectorRef: 'test',
        region: RUNTIME_INPUT_VALUE,
        roleArn: RUNTIME_INPUT_VALUE,
        stackName: RUNTIME_INPUT_VALUE
      }
    }
  }
}

const template = (spec: any): any => ({
  identifier: 'test_identifier',
  type: StepType.CloudFormationDeleteStack,
  timeout: RUNTIME_INPUT_VALUE,
  spec: {
    configuration: {
      type: StoreTypes.Inline,
      spec
    }
  }
})

const regionMock = {
  resource: [
    {
      name: 'GovCloud (US-West)',
      value: 'us-gov-west-1'
    },
    {
      name: 'GovCloud (US-East)',
      value: 'us-gov-east-1'
    }
  ]
}

const renderComponent = (data: any) => {
  return render(
    <TestWrapper
      path="/account/:accountId/cd/orgs/:orgIdentifier/projects/:projectIdentifier"
      pathParams={{ accountId: 'account', orgIdentifier: 'org', projectIdentifier: 'project' }}
    >
      <Formik initialValues={{}} onSubmit={() => undefined} formName="wrapperComponentTestForm">
        <FormikForm>
          <DeleteStackInputStepRef
            initialValues={initialValues as any}
            stepType={StepType.CloudFormationDeleteStack}
            stepViewType={StepViewType.InputSet}
            inputSetData={{
              template: template(data)
            }}
            path="test"
            allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.EXPRESSION]}
          />
        </FormikForm>
      </Formik>
    </TestWrapper>
  )
}

describe('Test cloudformation delete stack input set', () => {
  beforeAll(() => {
    jest
      .spyOn(cdServices, 'useGetConnector')
      .mockImplementation(() => ({ loading: false, error: null, data: {} } as any))
  })

  test('should render all input variables components', () => {
    jest
      .spyOn(Portal, 'useListAwsRegions')
      .mockImplementation(() => ({ loading: false, error: null, data: regionMock, refetch: jest.fn() } as any))
    const data = {
      provisionerIdentifier: RUNTIME_INPUT_VALUE,
      connectorRef: RUNTIME_INPUT_VALUE,
      region: RUNTIME_INPUT_VALUE,
      roleArn: RUNTIME_INPUT_VALUE,
      stackName: RUNTIME_INPUT_VALUE
    }
    const { container } = renderComponent(data)
    expect(container).toMatchSnapshot()
  })

  test('timeout should be updated', async () => {
    const data = {}
    const { container, getByPlaceholderText } = renderComponent(data)

    const timeoutInput = getByPlaceholderText('Enter w/d/h/m/s/ms')
    act(async () => {
      await userEvent.type(timeoutInput!, '10m')
    })

    await waitFor(() => expect(timeoutInput).toHaveDisplayValue('10m'))
    expect(container).toMatchSnapshot()
  })

  test('provisionerIdentifier should be updated', async () => {
    const data = {
      provisionerIdentifier: RUNTIME_INPUT_VALUE
    }
    const { container } = renderComponent(data)
    const provId = queryByAttribute('name', container, 'test.spec.configuration.spec.provisionerIdentifier')
    await act(async () => {
      await userEvent.type(provId!, 'testID')
    })
    expect(provId).toHaveDisplayValue('testID')
  })

  test('stack name should be updated', async () => {
    const data = {
      stackName: RUNTIME_INPUT_VALUE
    }
    const { container } = renderComponent(data)

    const stackName = queryByAttribute('name', container, 'test.spec.configuration.spec.stackName')
    await act(async () => {
      await userEvent.type(stackName!, 'testStackName')
    })

    expect(stackName).toHaveDisplayValue('testStackName')
    expect(container).toMatchSnapshot()
  })
})
