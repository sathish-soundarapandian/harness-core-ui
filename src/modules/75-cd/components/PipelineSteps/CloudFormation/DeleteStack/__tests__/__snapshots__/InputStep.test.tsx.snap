// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cloudformation delete stack input set should render all input variables components 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div>
        <form
          action="#"
          class="FormikForm--main"
        >
          <div>
            <div
              class="fieldAndOptions formGroup md"
            >
              <div
                class="bp3-form-group"
                data-id="timeout-0"
              >
                <label
                  class="bp3-label"
                  for="timeout"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="wrapperComponentTestForm_timeout"
                  >
                    pipelineSteps.timeoutLabel
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="container"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                      style="flex-grow: 1;"
                    >
                      <div
                        class="bp3-input-group bp3-fill"
                      >
                        <input
                          class="bp3-input"
                          name="timeout"
                          placeholder="Enter w/d/h/m/s/ms"
                          type="text"
                          value=""
                        />
                      </div>
                      <span
                        class="bp3-popover-wrapper MultiTypeInput--wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="MultiTypeInput--btn MultiTypeInput--FIXED"
                            data-testid="multi-type-button"
                          >
                            <span
                              class="bp3-icon StyledProps--main"
                              data-icon="fixed-input"
                            >
                              <svg
                                height="12"
                                viewBox="0 0 9 9"
                                width="12"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                  fill="currentColor"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="fieldAndOptions formGroup md"
            >
              <div
                class="bp3-form-group"
                data-id="test.spec.configuration.spec.provisionerIdentifier-1"
              >
                <label
                  class="bp3-label"
                  for="test.spec.configuration.spec.provisionerIdentifier"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="wrapperComponentTestForm_test.spec.configuration.spec.provisionerIdentifier"
                  >
                    pipelineSteps.provisionerIdentifier
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                  >
                    <div
                      class="bp3-input-group MultiTypeInput--input"
                    >
                      <input
                        autocomplete="off"
                        class="bp3-input"
                        name="test.spec.configuration.spec.provisionerIdentifier"
                        type="text"
                        value=""
                      />
                    </div>
                    <span
                      class="bp3-popover-wrapper MultiTypeInput--wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="MultiTypeInput--btn MultiTypeInput--FIXED"
                          data-testid="multi-type-button"
                        >
                          <span
                            class="bp3-icon StyledProps--main"
                            data-icon="fixed-input"
                          >
                            <svg
                              height="12"
                              viewBox="0 0 9 9"
                              width="12"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="formGroup md"
            >
              <div
                class="connectorLabel"
                style="margin-bottom: 10px;"
              >
                <div
                  class="StyledProps--font StyledProps--main"
                  data-test-id="connectorTooltip"
                >
                  <span
                    class="Tooltip--acenter bp3-label"
                    data-tooltip-id="wrapperComponentTestForm_test.spec.configuration.spec.connectorRef"
                  >
                    <p
                      class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey900"
                    >
                      pipelineSteps.awsConnectorLabel
                    </p>
                  </span>
                </div>
                <div
                  style="display: flex; align-items: center;"
                >
                  <div
                    class="bp3-form-group"
                    data-id="undefined-2"
                    style="margin-bottom: 0px;"
                  >
                    <div
                      class="bp3-form-content"
                    >
                      <div
                        class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                        style="width: 300px;"
                      >
                        <button
                          class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--iconOnly Button--without-shadow Button--withRightIcon"
                          data-testid="cr-field-test.spec.configuration.spec.connectorRef"
                          style="width: 300px;"
                          type="button"
                        >
                          <span
                            class="bp3-button-text"
                          >
                            <span
                              class="placeholder"
                            >
                              select
                            </span>
                          </span>
                          <span
                            class="bp3-icon bp3-icon-chevron-down StyledProps--main"
                            icon="chevron-down"
                          >
                            <svg
                              data-icon="chevron-down"
                              height="14"
                              viewBox="0 0 16 16"
                              width="14"
                            >
                              <desc>
                                chevron-down
                              </desc>
                              <path
                                d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                        <span
                          class="bp3-popover-wrapper MultiTypeInput--wrapper"
                        >
                          <span
                            class="bp3-popover-target"
                          >
                            <button
                              class="MultiTypeInput--btn MultiTypeInput--FIXED"
                              data-testid="multi-type-button"
                            >
                              <span
                                class="bp3-icon StyledProps--main"
                                data-icon="fixed-input"
                              >
                                <svg
                                  height="12"
                                  viewBox="0 0 9 9"
                                  width="12"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                    fill="currentColor"
                                  />
                                </svg>
                              </span>
                            </button>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="fieldAndOptions formGroup md"
            >
              <div
                class="bp3-form-group"
                data-id="test.spec.configuration.spec.region-3"
              >
                <label
                  class="bp3-label"
                  for="test.spec.configuration.spec.region"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="wrapperComponentTestForm_test.spec.configuration.spec.region"
                  >
                    regionLabel
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                  >
                    <div
                      class="bp3-popover-wrapper MultiTypeInput--select MultiTypeInput--fixedValueInput Select--main bp3-fill"
                    >
                      <div
                        class="bp3-popover-target"
                      >
                        <div
                          class="bp3-input-group"
                        >
                          <input
                            autocomplete="off"
                            class="bp3-input"
                            name="test.spec.configuration.spec.region"
                            placeholder="pipeline.regionPlaceholder"
                            style="padding-right: 0px;"
                            type="text"
                            value=""
                          />
                          <span
                            class="bp3-input-action"
                          >
                            <span
                              class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                              icon="chevron-down"
                            >
                              <svg
                                data-icon="chevron-down"
                                height="14"
                                viewBox="0 0 16 16"
                                width="14"
                              >
                                <desc>
                                  chevron-down
                                </desc>
                                <path
                                  d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                    <span
                      class="bp3-popover-wrapper MultiTypeInput--wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="MultiTypeInput--btn MultiTypeInput--FIXED"
                          data-testid="multi-type-button"
                        >
                          <span
                            class="bp3-icon StyledProps--main"
                            data-icon="fixed-input"
                          >
                            <svg
                              height="12"
                              viewBox="0 0 9 9"
                              width="12"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="fieldAndOptions formGroup md"
            >
              <div
                class="bp3-form-group"
                data-id="test.spec.configuration.spec.roleArn-4"
              >
                <label
                  class="bp3-label"
                  for="test.spec.configuration.spec.roleArn"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="wrapperComponentTestForm_test.spec.configuration.spec.roleArn"
                  >
                    connectors.awsKms.roleArnLabel
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                  >
                    <div
                      class="bp3-popover-wrapper MultiTypeInput--select MultiTypeInput--fixedValueInput Select--main bp3-fill"
                    >
                      <div
                        class="bp3-popover-target"
                      >
                        <div
                          class="bp3-input-group"
                        >
                          <input
                            autocomplete="off"
                            class="bp3-input"
                            name="test.spec.configuration.spec.roleArn"
                            placeholder="select"
                            style="padding-right: 0px;"
                            type="text"
                            value=""
                          />
                          <span
                            class="bp3-input-action"
                          >
                            <span
                              class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                              icon="chevron-down"
                            >
                              <svg
                                data-icon="chevron-down"
                                height="14"
                                viewBox="0 0 16 16"
                                width="14"
                              >
                                <desc>
                                  chevron-down
                                </desc>
                                <path
                                  d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                    <span
                      class="bp3-popover-wrapper MultiTypeInput--wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="MultiTypeInput--btn MultiTypeInput--FIXED"
                          data-testid="multi-type-button"
                        >
                          <span
                            class="bp3-icon StyledProps--main"
                            data-icon="fixed-input"
                          >
                            <svg
                              height="12"
                              viewBox="0 0 9 9"
                              width="12"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="fieldAndOptions formGroup md"
            >
              <div
                class="bp3-form-group"
                data-id="test.spec.configuration.spec.stackName-5"
              >
                <label
                  class="bp3-label"
                  for="test.spec.configuration.spec.stackName"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="wrapperComponentTestForm_test.spec.configuration.spec.stackName"
                  >
                    cd.cloudFormation.stackName
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                  >
                    <div
                      class="bp3-input-group MultiTypeInput--input"
                    >
                      <input
                        autocomplete="off"
                        class="bp3-input"
                        name="test.spec.configuration.spec.stackName"
                        type="text"
                        value=""
                      />
                    </div>
                    <span
                      class="bp3-popover-wrapper MultiTypeInput--wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="MultiTypeInput--btn MultiTypeInput--FIXED"
                          data-testid="multi-type-button"
                        >
                          <span
                            class="bp3-icon StyledProps--main"
                            data-icon="fixed-input"
                          >
                            <svg
                              height="12"
                              viewBox="0 0 9 9"
                              width="12"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </form>
  </div>
</div>
`;

exports[`Test cloudformation delete stack input set stack name should be updated 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div>
        <form
          action="#"
          class="FormikForm--main"
        >
          <div>
            <div
              class="fieldAndOptions formGroup md"
            >
              <div
                class="bp3-form-group"
                data-id="timeout-0"
              >
                <label
                  class="bp3-label"
                  for="timeout"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="wrapperComponentTestForm_timeout"
                  >
                    pipelineSteps.timeoutLabel
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="container"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                      style="flex-grow: 1;"
                    >
                      <div
                        class="bp3-input-group bp3-fill"
                      >
                        <input
                          class="bp3-input"
                          name="timeout"
                          placeholder="Enter w/d/h/m/s/ms"
                          type="text"
                          value=""
                        />
                      </div>
                      <span
                        class="bp3-popover-wrapper MultiTypeInput--wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="MultiTypeInput--btn MultiTypeInput--FIXED"
                            data-testid="multi-type-button"
                          >
                            <span
                              class="bp3-icon StyledProps--main"
                              data-icon="fixed-input"
                            >
                              <svg
                                height="12"
                                viewBox="0 0 9 9"
                                width="12"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                  fill="currentColor"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="fieldAndOptions formGroup md"
            >
              <div
                class="bp3-form-group"
                data-id="test.spec.configuration.spec.stackName-1"
              >
                <label
                  class="bp3-label"
                  for="test.spec.configuration.spec.stackName"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="wrapperComponentTestForm_test.spec.configuration.spec.stackName"
                  >
                    cd.cloudFormation.stackName
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                  >
                    <div
                      class="bp3-input-group MultiTypeInput--input"
                    >
                      <input
                        autocomplete="off"
                        class="bp3-input"
                        name="test.spec.configuration.spec.stackName"
                        type="text"
                        value="testStackName"
                      />
                    </div>
                    <span
                      class="bp3-popover-wrapper MultiTypeInput--wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="MultiTypeInput--btn MultiTypeInput--FIXED"
                          data-testid="multi-type-button"
                        >
                          <span
                            class="bp3-icon StyledProps--main"
                            data-icon="fixed-input"
                          >
                            <svg
                              height="12"
                              viewBox="0 0 9 9"
                              width="12"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </form>
  </div>
</div>
`;

exports[`Test cloudformation delete stack input set timeout should be updated 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      action="#"
      class="FormikForm--main"
    >
      <div>
        <form
          action="#"
          class="FormikForm--main"
        >
          <div>
            <div
              class="fieldAndOptions formGroup md"
            >
              <div
                class="bp3-form-group"
                data-id="timeout-0"
              >
                <label
                  class="bp3-label"
                  for="timeout"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="wrapperComponentTestForm_timeout"
                  >
                    pipelineSteps.timeoutLabel
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="container"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                      style="flex-grow: 1;"
                    >
                      <div
                        class="bp3-input-group bp3-fill"
                      >
                        <input
                          class="bp3-input"
                          name="timeout"
                          placeholder="Enter w/d/h/m/s/ms"
                          type="text"
                          value="10m"
                        />
                      </div>
                      <span
                        class="bp3-popover-wrapper MultiTypeInput--wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="MultiTypeInput--btn MultiTypeInput--FIXED"
                            data-testid="multi-type-button"
                          >
                            <span
                              class="bp3-icon StyledProps--main"
                              data-icon="fixed-input"
                            >
                              <svg
                                height="12"
                                viewBox="0 0 9 9"
                                width="12"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                  fill="currentColor"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </form>
  </div>
</div>
`;
